homeassistant/components/velbus/__init__.py

```diff
@@ -194,3 +194,21 @@ async def async_remove_entry(hass: HomeAssistant, entry: ConfigEntry) -> None:
         shutil.rmtree,
         hass.config.path(STORAGE_DIR, f"velbuscache-{entry.entry_id}"),
     )
+
+
+async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Migrate old entry."""
+    _LOGGER.debug("Migrating from version %s", config_entry.version)
+    cache_path = hass.config.path(STORAGE_DIR, f"velbuscache-{config_entry.entry_id}/")
+    if config_entry.version == 1:
+        # This is the config entry migration for adding the new program selection
+        # clean the velbusCache
+        if os.path.isdir(cache_path):
+            await hass.async_add_executor_job(shutil.rmtree, cache_path)
```

Comment:
We need to patch the directory check and `shutil.rmtree` to test line 207 too.

Reply:
i gave it a try, not sure if this is ok